### DISCUSSION TITLE
Measuring fire size from bounding box

### DISCUSSION THREAD

**Discussion by sajontahsen** [akdasUAF/ForestFireDetection#20] [Q&A] [answered]
Hello! Looking for some directions on the following:
1. Given the bounding boxes of the flame, what would be the preferred way to calculate its size?  My initial guess was to compute width*height of the box, but that has a dependency on the image size. 
2. Is deriving the actual area of the fire on the ground a goal of this project, or are we to primarily consider relative changes?

I have forked this repo, set up a simple script to read frames from video and ran the included YOLO model on the [(2) Raw video from Zenmuse X4S cameras for one specific pile](https://ieee-dataport.org/open-access/flame-dataset-aerial-imagery-pile-burn-detection-using-drones-uavs) from the FLAME dataset. 

Here's the script: [Github Link](https://github.com/LostSputnik/ForestFireDetectionTestRun/blob/dcc040ebfe62d6f4b87860e94c9b8d680e7bc346/FireTracking/track.py)
And the demo video showcasing the detections: [Drive Link](https://drive.google.com/file/d/1WZbm_cbxfFRdoSRLDVXDtu1uhApqX-Lg/view?usp=sharing)

My next goal is to plot the relative changes from the normalized width and height of the bounding boxes, pending further clarifications. I would also like to work on improving the model for the detection. Since this is primarily a prototype I'm working on, I would greatly appreciate any feedback. Am I on the right track?

Best regards,
Tahsen Islam Sajon 
[Resume](https://drive.google.com/file/d/1nUR4WYFnh4pY0yUSeteKR8PqZHS_9xgh/view?usp=drivesdk)  &#124; [Email](mailto:sajon.tahsen@gmail.com)  &#124; [LinkedIn](https://www.linkedin.com/in/tahsen-islam-sajon)

**Comment by YaliWang2019** [comment, marked as answer]
Hi Tahsen,

Yes, calculating the size of the bounding box is using its width * height, while we need to exclude the overlaying boxes. So far, we only need to consider the sizes on the image, not the actual fire sizes on the ground. I believe you are on the right track and you already had a good start here, just the YOLO model you trained so far might need some improvements. I've trained several YOLO models before, and they marked many overlapped boxes. When calculating the box sizes, the overlapped boxes need to be excluded, that's saying we're calculating the area only once.

Thanks,
Yali Wang

**Reply by sajontahsen** [reply]
Thank you very much for the detailed answer and positive feedback. I'll start working on calculating the size of the bounding box and plotting the changes over time. 

And yes, I would love to train a better model. I read from [this discussion](https://github.com/akdasUAF/ForestFireDetection/discussions/16#discussioncomment-8598104) that you are working on providing a fully annotated dataset based on the IEEE FLAME dataset for training. I think that'll be very useful. Also, there are segmentation masks provided in the FLAME dataset, and we can try training a Segment Anything Model or another and calculating the area from that. 

Should i continue this discussion in this thread? What would be your preferred medium for communication @YaliWang2019?

**Reply by YaliWang2019** [reply]
So far the discussion board is where we do most of the communications (and in this way we can all engage in the discussion). Also, since it seems like you already have a clear thought in mind, I’ll highly recommend you to start working on a proposal first and submit it once the application is open to the contributors. We’re excited and looking forward to reading your proposal : )

**Reply by sajontahsen** [reply]
All right, I'll start working on the proposal draft. The application period for GSoC'24 opens on March 19. Meanwhile, I would greatly appreciate any feedback you might have to enhance the proposal. 
Thank you once again, and I hope you have a wonderful day!

**Reply by khetansarvesh** [reply]
Since an object detection model is detecting multiple overlapping bounding boxes, we can also try framing this as a classification + localization problem statement right? In this case, only a single bounding box will get detected all the time and yes a fully annotated dataset would be lovely to have to finetune the foundations model for our specific use case!!!

**Reply by Vikas-Rajpurohit** [reply]
But what if we have the below frame as input, classification + localization will not be able to give information on
how many fire areas are detected also for calculating severity we will not have the bounding boxes.
![image](https://github.com/akdasUAF/ForestFireDetection/assets/112104373/fb33cf53-f9b2-46ca-a0d6-bbc0bf2f8ad1)

Also, @YaliWang2019 my proposed approach of calculating severity using fire pixels in the discussion [here](https://github.com/akdasUAF/ForestFireDetection/discussions/29#discussion-6364221) is it on right track?

**Reply by sajontahsen** [reply]
@YaliWang2019 Hello again, it's been a while 🙋‍♂️ I've submitted my proposal draft, and am looking forward to hearing your feedback. Thanks! Have a good day.
